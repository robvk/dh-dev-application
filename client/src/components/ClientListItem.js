import * as React from "react";
import classNames from "classnames";

import "./ClientListItem.css";

type ClientList = {
  origin: string,
  first_name: string,
  last_name: string
};

type Props = {
  client: ClientList,
  isSelected: boolean
};

function ClientListItem(props: Props) {
  const { client, isSelected } = props;
  const { first_name, last_name, origin } = client;

  return (
    <div
      className={classNames("client-dropdown-list-item", {
        "client-dropdown-list-item--selected": isSelected
      })}
    >
      {first_name} {last_name} from {origin}
    </div>
  );
}

export default ClientListItem;
