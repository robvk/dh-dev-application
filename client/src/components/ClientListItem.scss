@import "../appColors";

.client-dropdown-list-item {
  padding: 10px;
}

.client-dropdown-list-item:hover,
.client-dropdown-list-item--selected {
  background-color: $primary-color;
  color: $background-color;
}
