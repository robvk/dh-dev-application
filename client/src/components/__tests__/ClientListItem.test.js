import React from "react";
import ReactDOM from "react-dom";

import ClientListItem from "../ClientListItem";

const testClient = {
  id: "59a2afbbfc13ae278700044c",
  first_name: "Rochella",
  last_name: "Chipchase",
  email: "rchipchase0@joomla.org",
  gender: "Female",
  photo: "https://robohash.org/molestiaedoloremquesed.jpg?size=50x50&set=set1",
  origin: "Egypt"
};

it("renders correctly", () => {
  const wrapper = shallow(
    <ClientListItem client={testClient} isSelected={false} />
  );
  expect(wrapper).toMatchSnapshot();
});

it("adds css modifier if selected", () => {
  const wrapper = shallow(<ClientListItem client={testClient} isSelected={true} />);
  expect(wrapper.is(".client-dropdown-list-item--selected")).toBe(true);
});
