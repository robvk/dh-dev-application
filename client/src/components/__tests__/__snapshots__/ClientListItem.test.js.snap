// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="client-dropdown-list-item"
>
  Rochella
   
  Chipchase
   from 
  Egypt
</div>
`;
